py/gc.c

```diff
@@ -227,43 +272,33 @@ STATIC void gc_sweep(void) {
     MP_STATE_MEM(gc_collected) = 0;
     #endif
     // free unmarked heads and their tails
-    int free_tail = 0;
-    for (mp_uint_t block = 0; block < MP_STATE_MEM(gc_alloc_table_byte_len) * BLOCKS_PER_ATB; block++) {
-        switch (ATB_GET_KIND(block)) {
-            case AT_HEAD:
+    for (mp_uint_t block = mem_first(); block != MEM_BLOCK_ERROR; block = mem_next(block)) {
+        if (ATB_GET_KIND(block) == AT_MARK){
+            ATB_MARK_TO_HEAD(block);
+        } else {
+            // TODO: bad indentation kept intentionally for git diff (will move back next commit)
```

Comment:
Such things should not be necessary: use ```git diff -w``` to ignore whitespace. Or for the diff on githube: append `?w=` to the end of this url. That is invalueable anyway fot commits like this else all you see is one huge bunch of adds/deletes.